dotCMS/src/main/java/com/dotcms/saml/DotSamlProxyFactory.java

```diff
@@ -0,0 +1,215 @@
+package com.dotcms.saml;
+
+import com.dotcms.osgi.OSGIConstants;
+import com.dotcms.security.apps.AppDescriptor;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.util.CollectionsUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+import io.vavr.control.Try;
+import org.apache.felix.framework.OSGIUtil;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * This is the proxy to provides the object to interact with the Saml Osgi Bundle
+ *
+ * @author jsanca
+ */
+public class DotSamlProxyFactory {
+
+    public static final String SAML_APP_CONFIG_KEY = "dotsaml-config";
+    public static final String PROPERTIES_PATH = File.separator + "saml" + File.separator + "dotcms-saml-default.properties";
+
+    private static final String ASSETS_PATH = Config.getStringProperty("ASSET_REAL_PATH",
+            FileUtil.getRealPath(Config.getStringProperty("ASSET_PATH", "/assets")));
+    private static final String IDP_FILE_PATH = ASSETS_PATH + PROPERTIES_PATH;
+
+    private final MessageObserver    messageObserver    = new DotLoggerMessageObserver();
+    private final AppsAPI            appsAPI            = APILocator.getAppsAPI();
+    private final IdentityProviderConfigurationFactory identityProviderConfigurationFactory =
+            new DotIdentityProviderConfigurationFactoryImpl(this.appsAPI, APILocator.getHostAPI());
+
+    private SamlServiceBuilder        samlServiceBuilder         = null;
```

Comment:
Codacy found an issue: [Avoid using redundant field initializer for 'samlServiceBuilder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)